numbat/modules/math/combinatorics.nbt

```diff
@@ -0,0 +1,34 @@
+use core::error
+use core::functions
+use math::transcendental
+
+@name("Factorial")
+@description("The product of the integers 1 through n, also written n!")
+@url("https://en.wikipedia.org/wiki/Factorial")
+fn factorial(n: Scalar) -> Scalar = n!
+
+@name("Falling factorial")
+@description("Equal to n⋅(n-1)⋅…⋅(n-k+2)⋅(n-k+1) (k terms total). If n is an integer, this is the number
+	of k-element permutations from a set of size n. k must always be an integer.")
```

Comment:
Does this whitespace lead to problems in the rendered output? I think you can have multiple `@description` fields and they will be concatenated.

Reply:
Multiple `@description` fields are concatenated with newlines between them, which I'm not sure I want. But then again the literal newline is picked up in the string, so there may not be a way to get this stored as a single line while writing it across multiple lines.

Reply:
Separately, this has me wondering whether there would be merit in having separate `@description`s for docs and for numbat CLI. I would like to write that product as `$n\cdot(n-1)\cdot\ldots\cdot(n-k+2)\cdot(n-k+1)$` for the docs, but I wouldn't want to have to read that in the CLI. (Is there a LaTeX-to-unicode compiler?)

Reply:
We discussed this partially in #512 but didn't come to a satisfactory conclusion so far. I think the best to do as of now would be to directly use the Unicode characters that would be produced during compilation in math blocks instead of their LaTeX commands (e.g `⋅` instead of `\cdot`, `…` for `\ldots` and so on). MathJax handles this well as far as I can tell and it is also decently readable as plain text.